src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -91,6 +100,13 @@ func (c *IdleTimingConn) TimesOutAt() time.Time {
 
 // Read implements the method from io.Reader
 func (c *IdleTimingConn) Read(b []byte) (int, error) {
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
```

Comment:
The key thing is that the resource we're protecting is not the closed status, it's the underlying connection.